massa-signature/src/signature_impl.rs

```diff
@@ -722,92 +722,68 @@ impl<'de> ::serde::Deserialize<'de> for Signature {
-    /// let data = Hash::compute_from("Hello World!".as_bytes());
-    /// let signature = sign(&data, &private_key).unwrap();
-    /// let verification: bool = verify_signature(&data, &signature, &public_key).is_ok();
-    /// ```
-    fn verify(
-        &self,
-        hash: &Hash,
-        signature: &Signature,
-        public_key: &PublicKey,
-    ) -> Result<(), MassaHashError> {
-        let message = Message::from_slice(&hash.to_bytes())?;
-        Ok(self.0.verify(&message, &signature.0, &public_key.0)?)
-    }
-}
-
-/// Generate a random private key from a RNG.
-pub fn generate_random_private_key() -> PrivateKey {
-    use secp256k1::rand::rngs::OsRng;
-    let mut rng = OsRng::new().expect("OsRng");
-    PrivateKey(secp256k1::key::SecretKey::new(&mut rng))
-}
-
+/// Derives a PublicKey from a PrivateKey.
+///
+/// # Example
+/// ```
+/// # use massa_signature::{derive_public_key, generate_random_private_key};
+/// # use serde::{Deserialize, Serialize};
+/// let private_key = generate_random_private_key();
+/// let public_key = derive_public_key(&private_key);
+/// ```
 pub fn derive_public_key(private_key: &PrivateKey) -> PublicKey {
-    SIGNATURE_ENGINE.with(|signature_engine| signature_engine.derive_public_key(private_key))
+    PublicKey(private_key.0.public_key())
 }
 
+/// Returns the Signature produced by signing
+/// data bytes with a PrivateKey.
+///
+/// # Example
+///  ```
+/// # use massa_signature::{derive_public_key, generate_random_private_key, sign, PublicKey};
+/// # use massa_hash::hash::Hash;
+/// # use serde::{Deserialize, Serialize};
+/// let private_key = generate_random_private_key();
+/// let public_key: PublicKey = derive_public_key(&private_key);
+/// let data = Hash::compute_from("Hello World!".as_bytes());
+/// let signature = sign(&data, &private_key).unwrap();
+/// ```
 pub fn sign(hash: &Hash, private_key: &PrivateKey) -> Result<Signature, MassaHashError> {
-    SIGNATURE_ENGINE.with(|signature_engine| signature_engine.sign(hash, private_key))
+    let message = Message::from_slice(&hash.to_bytes())?;
+    Ok(Signature(SECP256K1.sign_schnorr(&message, &private_key.0)))
 }
 
+/// Checks if the Signature associated with data bytes
+/// was produced with the PrivateKey associated to given PublicKey
+///
+/// # Example
+///  ```
+/// # use massa_signature::{derive_public_key, generate_random_private_key, sign, verify_signature, PublicKey};
+/// # use massa_hash::hash::Hash;
+/// # use serde::{Deserialize, Serialize};
+/// let private_key = generate_random_private_key();
+/// let public_key: PublicKey = derive_public_key(&private_key);
+/// let data = Hash::compute_from("Hello World!".as_bytes());
+/// let signature = sign(&data, &private_key).unwrap();
+/// let verification: bool = verify_signature(&data, &signature, &public_key).is_ok();
+/// ```
 pub fn verify_signature(
     hash: &Hash,
     signature: &Signature,
     public_key: &PublicKey,
 ) -> Result<(), MassaHashError> {
```

Comment:
it's a bit strange that we return an error type named "MassaHashError" on a signature function